### ISSUE TITLE
Accessibility: Add alt attribute to hero img

### ISSUE THREAD

**Issue by aparkening** [alistapart/AListApart#215]
From Andrew's accessibility audit:

We use alt=”” on the main image for articles, but this is clearly more than decorative content.   Example: img src="/d/_made/d/ALA_417_Initiation-to-Code_960_379_81.jpg" alt=""

We could also use longdesc and write a lengthy description for the image but then would need to host an additional page for that content.  I’m not sure that this is worth it though.

This is a WCAG 2.0 1.1.1 failure.

**Comment by murtaugh** [comment]
@awkawk  The example image is missing in this post, which is too bad because I'd like to see an example of an illustration that is more that decorative. I'm not sure I can think of one that *I* think of as more than decorative, but you certainly have a different perspective.

**Comment by aparkening** [comment]
@murtaugh: I included the img call originally, but Github stripped it out.  I've re-added it to the original post.

**Comment by aparkening** [comment]
@awkawk I'm on the fence as to whether the main article image is decorative or not.  As a test on whether these are decorative or not, can you write useful image descriptions for 3 articles?  

If you, as the accessibility expert, find it really difficult to write useful descriptions, then I think we shouldn't describe them.

**Comment by murtaugh** [comment]
I think the thing I don't know, due to my own ignorance, is whether screen reader users like/expect a description like "an illustration of a man on a beach reading under a smiling umbrella."

**Comment by awkawk** [comment]
Some will want/expect it, others won't care.

"illustration of a man on a beach reading under a smiling umbrella" is perfect for that one as far as I'm concerned.  Others are harder.

http://alistapart.com/article/container-queries-once-more-unto-the-breach:
An illustration of a team of mice removing cheese from a mousetrap

http://alistapart.com/article/understanding-emotional-response
Illustration of a collection of tea kettles with different shapes and designs, all vigorously expelling steam

http://alistapart.com/article/risky-business-of-onboarding: yikes
http://alistapart.com/article/performance-showing-versus-telling: Yikes.

I'd say that the ideal solution would be to ask the illustrators to write a short description that they feel conveys the content sufficiently without giving away any joke that they are embedding.  (e.g. "Tea kettles boiling over.  This is funny because your users are as different as these kettles!  Ha ha!")

**Comment by awkawk** [comment]
Or, we could do alt="".  either is acceptable, omitting the alt as we do now is what isn't ok.

**Comment by mikepick** [comment]
Chiming in though my opinion isn't important—I love the description idea, it has potential to be quirky or surreal and funny, which is a tonal element the illustration provides that screen readers are not getting.

We could ask illustrators for a one-sentence description going forward, if we want to split the difference.

**Comment by murtaugh** [comment]
@awkawk where are we omitting the alt attribute? the original issue was about empty alt tags. Or do you mean there's a difference between `alt` and `alt=""`?

**Comment by murtaugh** [comment]
@mikepick it would be super easy to add a field for that, and if empty alt attributes are acceptable, either populate it or not.

**Comment by awkawk** [comment]
I guess I didn't re-read my initial comment well enough.  We _do_ use alt="" now but I was saying that the image is non-decorative and we should have a non-null alt.  I still agree with that.

That said, it is also really hard to write good alt descriptions for some of the images, so if we can ask for the description and make a good-faith effort to get the text and populate the alt when we have it we will be doing the right thing.

**Comment by murtaugh** [comment]
Sounds good. We'll set up a field for the description, @mikepick if you could add a step in the process to collect, those, if possible,and we'll be good.
